pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -484,56 +486,86 @@ def _check_and_set_fh(
-        (1) 1 if seasonality is not detected at the derived seasonal period, or
-        (2) the derived seasonal period if seasonality is detected at that value.
+        After deriving the seasonal periods, a seasonality test is performed for each
+        value of seasonal_period. Final seasonal period class attribute value is set equal to
+        (1) 1 if seasonality is not detected at any of the derived seasonal periods, or
+        (2) the derived seasonal periods for which seasonality is detected.
 
         Parameters
         ----------
         data : pd.DataFrame
             Data used index can be used to extract the seasonal period information
-        seasonal_period : Optional[Union[int, str]]
+        seasonal_period : Optional[Union[List[Union[int, str]], int, str]]
             Seasonal Period specified by user
 
         Raises
         ------
         ValueError
-            If seasonal period is provided but is not of type int or string
+            If seasonal period is provided but its values are not of type int or string
         """
         self.logger.info("Set up Seasonal Period.")
 
         # sktime is an optional dependency
         from sktime.utils.seasonality import autocorrelation_seasonality_test
 
         if seasonal_period is None:
-            index_freq = data.index.freqstr
-            self.seasonal_period = get_sp_from_str(str_freq=index_freq)
-        else:
-            if not isinstance(seasonal_period, (int, str)):
-                raise ValueError(
-                    f"seasonal_period parameter must be an int or str, got {type(seasonal_period)}"
-                )
+            seasonal_period = data.index.freqstr
 
-            if isinstance(seasonal_period, str):
-                self.seasonal_period = get_sp_from_str(str_freq=seasonal_period)
-            else:
-                self.seasonal_period = seasonal_period
+        if not isinstance(seasonal_period, list):
+            seasonal_period = [seasonal_period]
+        seasonal_period = [self._convert_sp_to_int(sp) for sp in seasonal_period]
 
         # check valid seasonal parameter
-        self.seasonality_present = autocorrelation_seasonality_test(
-            data[self.target_param], self.seasonal_period
-        )
+        seasonality_test_results = [
+            autocorrelation_seasonality_test(data[self.target_param], sp)
+            for sp in seasonal_period
+        ]
+        self.seasonality_present = any(seasonality_test_results)
+        sp_values_and_test_result = zip(seasonal_period, seasonality_test_results)
 
         # What seasonal period should be used for modeling?
-        self.sp_to_use = self.seasonal_period if self.seasonality_present else 1
+        self.all_sp_values = [
+            sp
+            for sp, seasonality_present in sp_values_and_test_result
+            if seasonality_present
+        ] or [1]
+        self.sp_to_use = self.all_sp_values[0]
+        self.seasonal_period = seasonal_period[0]
```

Comment:
Makes sense. I wasn't sure at that moment if `seasonal_period` was in use somewhere else and thus, I kept it as a single value. I have double checked now and yes, it seems we can do so.

Just in case, I still turn it into an scalar value if the `seasonal_period` list has only one element for retrocompatibility with the users (e.g., somebody using `exp.seasonal_period`  and suddenly receiving a list of one element instead of a value).